data_model/src/trigger/set.rs

```diff
@@ -273,29 +273,37 @@ impl Set {
     /// This actions will be inspected in the next [`Set::handle_data_event()`] call
     // Passing by value to follow other `handle_` methods interface
     #[allow(clippy::needless_pass_by_value)]
-    pub fn handle_data_event(&self, event: DataEvent) {
+    pub fn handle_data_event(&mut self, event: DataEvent) {
+        let mut acc = Vec::new();
         for entry in self.data_triggers.iter() {
             let id = entry.key();
 
             if id.domain_id.is_some() && id.domain_id.as_ref() != event.domain_id() {
                 continue;
             }
 
-            self.match_and_insert_trigger(event.clone(), entry.pair());
+            acc.push((event.clone(), (entry.key().clone(), entry.value().clone())));
+        }
+        for (event, pair) in acc {
+            self.match_and_insert_trigger(event, pair);
         }
     }
 
     /// Handle [`ExecuteTriggerEvent`].
     ///
     /// Finds all actions, that are triggered by `event` and stores them.
     /// This actions will be inspected in the next [`Set::inspect_matched()`] call
-    pub fn handle_execute_trigger_event(&self, event: ExecuteTriggerEvent) {
-        let entry = match self.by_call_triggers.get(&event.trigger_id) {
-            Some(entry) => entry,
-            None => return,
-        };
+    pub fn handle_execute_trigger_event(&mut self, event: ExecuteTriggerEvent) {
+        let pair;
+        {
+            let entry = match self.by_call_triggers.get(&event.trigger_id) {
+                Some(entry) => entry,
+                None => return,
+            };
+            pair = (entry.key().clone(), entry.value().clone());
```

Comment:
```suggestion
        let pair = {
            let entry = match self.by_call_triggers.get(&event.trigger_id) {
                Some(entry) => entry,
                None => return,
            };
            (entry.key().clone(), entry.value().clone())
```

Reply:
In rust everything is an expression so you can do the assignment without duplicating the name.

Reply:
yes, I think I wrote this a while ago before I had gotten used to the rust style.